components/fxa-client/android/src/main/java/mozilla/appservices/fxa/Devices.kt

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.fxa
+
+import mozilla.appservices.fxaclient.Device
+import mozilla.appservices.fxaclient.DeviceCapability
+import mozilla.appservices.fxaclient.DevicePushSubscription
+import mozilla.appservices.fxaclient.IncomingDeviceCommand
+import mozilla.appservices.sync15.DeviceType
+
+/**
+ * Manages the list of connected devices for a [FirefoxAccount]
+ */
+abstract class DeviceManager {
+    /**
+     * Set name of the current device.
+     * @param name New device name.
+     * @throwns FxaException.Authentication
+     */
+    abstract suspend fun setDeviceName(name: String)
+
+    /**
+     * Get the current device list. May be incomplete if state was never queried.
+     * @return [DeviceList] describes current and other known devices
+     */
+    abstract fun getKnownDevices(): DeviceList
+
+    /**
+     * Set a [DevicePushSubscription] for the current device.
+     * @param subscription A new [DevicePushSubscription].
+     * @throwns FxaException.Authentication
+     */
+    abstract suspend fun setDevicePushSubscription(subscription: DevicePushSubscription)
+
+    /**
+     * Send a command to a specified device.
+     * @param targetDeviceId A device ID of the recipient.
+     * @param title tab title
+     * @param url tab URL
+     * @throwns FxaException.Authentication
+     */
+    abstract suspend fun sendSingleTab(targetDeviceId: String, title: String, url: String)
+
+    /**
+     * Refreshes [DeviceList]. Registered [AccountEventsObserver] observers will be notified.
+     *
+     * @throwns FxaException.Authentication
+     */
+    abstract suspend fun refreshDevices()
```

Comment:
The abstract functions don't make this clear, but I want most of these methods should have some common functionality, probably achieved by some sort of wrapper:

 - If they alter the internal state, then `StorageHandler.saveState()` should be called (similar to [this android-components code](https://github.com/mozilla-mobile/firefox-android/blob/main/android-components/components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/FirefoxAccount.kt#L44-L65))
 - If we see an error with one of our auth tokens, then the internal state should be changed to `AccountState.Disconnected`, and `AccountEvent.AccountStateChanged()` should be sent out.